docs/resources/identity_provider.md

```diff
@@ -0,0 +1,191 @@
+---
+subcategory: "Identity and Access Management (IAM)"
+---
+
+# huaweicloud_identity_provider
+
+Manages the identity providers within HuaweiCloud IAM service.
+
+-> **NOTE:** You can create up to 10 identity providers.
+
+## Example Usage
+
+### Create a SAML protocol provider
+
+```hcl
+resource "huaweicloud_identity_provider" "provider_1" {
+  name     = "example_com_provider_saml"
+  protocol = "saml"
+
```

Comment:
`metadata` field is missing

Reply:
`metadata` is optional, I did not add it to make the sample clear